### PR TITLE
Chore: Handle public key revocation time for Manifests

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

After investigating what actually happens when the GPG key used for signing plugins gets revoked, I verified that the `opengpg` library rejects a signature if the key has been revoked, regardless of the revocation time. The problem with this is that if we need to revoke a key, **_all_** plugins would need to be re-signed and re-published, unless the user allow invalid signatures for them.

In this PR, I added a check that verifies that if the plugin was signed before the key was revoked, we still assume that the key is valid (logging a warning). This way, the revocation will be transparent for the final user and once a plugin gets updated, it will use the new key.

This has a caveat though. If the signature time can be tampered with (I don't know if that possible) or if there is a gap between the time the key is leaked and whenever it is revoked, there could be cases in which we allow a non-legit signature.

Thoughts?

**Which issue(s) does this PR fix?**:

Ref #23619

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -338,7 +339,7 @@ require (
 	github.com/AzureAD/microsoft-authentication-library-for-go v0.7.0 // indirect
 	github.com/Masterminds/sprig/v3 v3.2.2
 	github.com/Microsoft/go-winio v0.6.0 // indirect
-	github.com/ProtonMail/go-crypto v0.0.0-20210428141323-04723f9f07d7
+	github.com/ProtonMail/go-crypto v0.0.0-20230426101702-58e86b294756
```
Comment (andresmgot) [inline go.mod:342]: with this update, I was able to properly use errors returned by the `opengpg` library and compare them with  `errors.Is(err, ErrRevokedKey)`

File: pkg/plugins/manager/signature/manifest.go [hunk 2]
```diff
@@ -339,15 +339,46 @@ func (s *Signature) Verify(ctx context.Context, keyID string, block *clearsign.B
...
+
 	if _, err = openpgp.CheckDetachedSignature(keyring,
 		bytes.NewBuffer(block.Bytes),
-		block.ArmoredSignature.Body, &packet.Config{}); err != nil {
-		// If the key includes revocations, we can assume that the key was revoked
-		if len(keyring) > 0 && len(keyring[0].Revocations) > 0 {
-			return fmt.Errorf("%s (KeyID: %s): %w", openpgpErrors.ErrKeyRevoked.Error(), keyID, err)
+		bytes.NewBuffer(signatureBody), &packet.Config{}); err != nil {
+		if errors.Is(err, openpgpErrors.ErrKeyRevoked) {
+			valid, err := wasSignedBeforeRevocation(keyring, signatureBody)
```
Comment (marefr) [inline pkg/plugins/manager/signature/manifest.go:351]: Nit. Easier to read perhaps.

```suggestion
			signedBeforeRevocation, err := wasSignedBeforeRevocation(keyring, signatureBody)
```

**Review by marefr — COMMENTED** [review]
Looks good in general. Added some comments/suggestions. As discussed letting security have a look at this is probably a good idea.

**Comment by andresmgot** [comment]
After checking with the security team, we decided to not pursuit this solution since there is no proof that the timestamp of the signed MANIFEST.txt cannot be tampered with, therefore bypassing the signature check.

I will open another PR with just the update of the crypto library since it improves how we check if the key has been revoked.

**Pull request closed without merging**